pkg/subvolume/subvolume.go

```diff
@@ -0,0 +1,122 @@
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+)
+
+func List(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace, operatorNamespace string) {
+
+	getpv := `kubectl get pv -o jsonpath={.items[*].spec.csi.volumeAttributes.subvolumeName}`
+	// Get all csi reference on k8s side for all cephfs pv
+	k8sRef := exec.ExecuteBashCommand(getpv)
+	if k8sRef == "" {
+		logging.Fatal(fmt.Errorf("No k8s pvs found. Nothing to do."))
+	}
+	k8sRefl := strings.Split(k8sRef, " ")
+	//logging.Info("k8s Ref found:%s", k8sRefl)
+
+	// get the csi reference on CSI side for all cephfs PV
+	svCSIRef := getCSIRef(ctx, clientsets, clusterNamespace, operatorNamespace)
+
+	// Check whether difference is of the two lists is measurable
+	diff := getDifference(svCSIRef, k8sRefl)
+	if diff == nil {
+		logging.Info("No stale volumes found")
+	}
+	fmt.Println("list of stale volumes to be deleted", diff)
+}
+
+func getDifference(csiref, k8ref []string) []string {
+	mb := make(map[string]struct{}, len(k8ref))
+	for _, x := range k8ref {
+		mb[x] = struct{}{}
+	}
+	var diff []string
+	for _, x := range csiref {
+		if _, found := mb[x]; !found {
+			// reference present in ceph but not in k8s
+			diff = append(diff, x)
+		}
+	}
+	if diff == nil {
+		logging.Fatal(fmt.Errorf("No stale Volume found"))
+	}
+
+	return diff
+}
+
+func getCSIRef(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace, operatorNamespace string) []string {
+
+	fsList := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "ls", "--format", "json"}, clusterNamespace, operatorNamespace, true, false)
+	if fsList == "" {
+		logging.Fatal(fmt.Errorf("failed to get filesystem"))
+	}
+	type fsStruct struct {
+		Name string
+		data []string
+	}
+	var fsstruct []fsStruct
+	var subvolg []fsStruct
+	var subvol []fsStruct
+
+	err := json.Unmarshal([]byte(fsList), &fsstruct)
+	if err != nil {
+		logging.Fatal(err)
+	}
+
+	var svList string
+
+	for _, fs := range fsstruct {
+		//logging.Info("getting subvolume for filesystem %s", fs.Name)
+		svgList := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolumegroup", "ls", fs.Name, "--format", "json"}, clusterNamespace, operatorNamespace, true, false)
+		if svgList == "" {
+			logging.Fatal(fmt.Errorf("failed to get subvolumegroup"))
+		}
+		err := json.Unmarshal([]byte(svgList), &subvolg)
+		if err != nil {
+			logging.Fatal(err)
+		}
+		for _, svg := range subvolg {
+			//logging.Info("getting subvolume for subvolumegroup %s", svg.Name)
+			// Get all csi reference on ceph side for all subvolume
+			svList = exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolume", "ls", fs.Name, svg.Name}, clusterNamespace, operatorNamespace, true, false)
+			if svList == "" {
+				logging.Info("No subvolume found for fs %s svg %s", fs, svg)
+			}
+		}
+	}
+	errg := json.Unmarshal([]byte(svList), &subvol)
```

Comment:
```suggestion
	errg := json.Unmarshal([]byte(svList), &subvol)
```
create helper function for unMarshal